### ISSUE TITLE
Relative import for proto package does not work

### ISSUE THREAD

**Issue by ElDavoo** [ElDavoo/wa-crypt-tools#63]
See Readme.  
Why do we have to replace relative import of proto files with absolute import?  
E.g. why do we have to replace:
`import C14_cipher_version_pb2 as C14__cipher__version__pb2`  
with 
`import wa_crypt_tools.C14_cipher_version_pb2 as C14__cipher__version__pb2`  
??  
This makes more difficult to automate proto class generation.

**Comment by Mikel12455** [comment]
To be completely honest, The only thing I knew a couple of days earlier is that those imports just didn't work.

Though I just tested something and they can be replaced with something like:

`from . import C14_cipher_version_pb2 as C14__cipher__version__pb2`

I don't know if this fixes the issue, so let me know!

**Comment by ElDavoo** [comment]
The issue is "find a way to generate proto classes from protoc that work". Fixing imports after generating the proto classes is already what i'm doing (see readme), but it's only a workaround.
The problem is:  if you use protoc to generate the python files, then importing will fail.

**Comment by Mikel12455** [comment]
> The problem is: if you use protoc to generate the python files, then importing will fail.

Oh okay, the problem is with `protoc`, not with its generated Python files. These might be useful then:

[Protocol Buffer issue](https://github.com/protocolbuffers/protobuf/issues/1491)

[Tool to fix imports](https://github.com/cpcloud/protoletariat)

The tool can be installed with pip, so it may become a dev dependency too if actually useful. Gonna test it right now.

**Comment by Mikel12455** [comment]
Run a couple of tests and it seems to work

**Comment by ElDavoo** [comment]
Ok, I will try to test when I have some time

**Comment by markdoerr** [comment]
Hi @ElDavoo 
using 'protoletariat' is a very ugly solution, why is the "relative import problem" not solved in the first place by simply generating the correct import syntax:

`from . import my_pb2 as ... `
instead of:
`import my_pb2 as ... ` ?

Very simple, no other tools required ...

**Comment by ElDavoo** [comment]
> why is the "relative import problem" not solved in the first place by simply generating the correct import synta

Because I'm a bad developer =)  

Seriously, there was a reason, but i completely forgot it. maybe it was a stupid reason.

**Comment by Mikel12455** [comment]
> Hi @ElDavoo, using 'protoletariat' is a very ugly solution, why is the "relative import problem" not solved in the first place by simply generating the correct import syntax:
> 
> `from . import my_pb2 as ... ` instead of: `import my_pb2 as ... ` ?
> 
> Very simple, no other tools required ...

How do you generate them like that? protoc always generates them like `import my_pb2 as ...`, so you have to manually fix the imports. protoletariat just does this automatically.

If having a new dependency seems overkill though, we can just revert to the manual fix and change the README.md a bit.

**Comment by ElDavoo** [comment]
Maybe they're talking about the imports in the main script

**Comment by markdoerr** [comment]
> > Hi @ElDavoo, using 'protoletariat' is a very ugly solution, why is the "relative import problem" not solved in the first place by simply generating the correct import syntax:
> > `from . import my_pb2 as ... ` instead of: `import my_pb2 as ... ` ?
> > Very simple, no other tools required ...
> 
> How do you generate them like that? protoc always generates them like `import my_pb2 as ...`, so you have to manually fix the imports. protoletariat just does this automatically.
> 

Who would be able to directly fix the import issue in the protoc compiler ? 

I think all three of us (@ElDavoo , @Mikel12455  and myself agree on the same solution) - do you know, whom we could address to fix the import generation in protoc ?

> If having a new dependency seems overkill though, we can just revert to the manual fix and change the README.md a bit.

Yes, indeed, @Mikel12455 - the problem should be fixed in the protoc compiler, not by an extra tool that is just made to fix a bug that should be easily solvable.

**Comment by markdoerr** [comment]
I raised the issue in the grpc / protoc community:

[https://github.com/grpc/grpc/issues/29459](https://github.com/grpc/grpc/issues/29459)

 Please feel free to support it there to increase attention for the issue.
Thanks.

**Comment by ElDavoo** [comment]
I think the revelant discussion is [this](https://github.com/protocolbuffers/protobuf/issues/1491). The root issue has already been linked by Mikel:

> [Protocol Buffer issue](https://github.com/protocolbuffers/protobuf/issues/1491)

I reopened the issue because I thought you had a better solution for this problem, but we can't make any further action in this repository now, so I will close it again.

We all know it's a hacky workaround, and before Mikel's commits I manually patched the files, but it's all we can do for now, unless the linked protobuf issue gets progress.
